### PR TITLE
Fix parameter missing when using OBO process

### PR DESCRIPTION
**Pull request by moarychan**
Fixes https://github.com/Azure/azure-sdk-for-java/issues/30359

Make the extra parameter converter `AadJwtBearerGrantRequestEntityConverter` work for JwtBearerOAuth2AuthorizedClientProvider no matter any scenario.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/configuration/AadOAuth2ClientConfiguration.java [hunk 1]
```diff
@@ -126,14 +126,13 @@ OAuth2AuthorizedClientManager authorizedClientManager(ClientRegistrationReposito
     JwtBearerOAuth2AuthorizedClientProvider azureAdJwtBearerProvider(ObjectProvider<OAuth2ClientAuthenticationJwkResolver> resolvers) {
         JwtBearerOAuth2AuthorizedClientProvider provider = new JwtBearerOAuth2AuthorizedClientProvider();
         OAuth2ClientAuthenticationJwkResolver resolver = resolvers.getIfUnique();
+        AadJwtBearerGrantRequestEntityConverter jwtBearerConverter = new AadJwtBearerGrantRequestEntityConverter();
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/configuration/AadOAuth2ClientConfiguration.java:129]: why moving this out will solve this issue?

Comment (moarychan) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/configuration/AadOAuth2ClientConfiguration.java:129]: this converter is an extension against `JwtBearerGrantRequestEntityConverter`, which is provided by default in Spring Security, the OBO flow requires an extra parameter to interact with Azure AD, so this converter will always be added in `JwtBearerOAuth2AuthorizedClientProvider`.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/aad/implementation/oauth2/AadOAuth2ClientConfigurationTests.java [hunk 2]
```diff
@@ -141,4 +151,105 @@ void testResourceServerWithOboExistCustomAndGraphClient() {
...
+
+                ClientAuthorizationException exception = null;
+                try {
+                    jwtBearerProvider.authorize(auth2AuthorizationContext);
+                } catch (ClientAuthorizationException ex) {
+                    exception = ex;
+                } finally {
+                    assertThat(exception).isNotNull();
+                    assertThat(exception.getMessage())
+                        .isEqualTo("[invalid_key] Failed to resolve JWK signing key for client registration 'graph'.");
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/aad/implementation/oauth2/AadOAuth2ClientConfigurationTests.java:198]: Why failed? Can we have a test case that use `private_key_jwt` successfully?

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/aad/implementation/oauth2/AadOAuth2ClientConfigurationTests.java [hunk 3]
```diff
@@ -141,4 +151,105 @@ void testResourceServerWithOboExistCustomAndGraphClient() {
...
+    @Test
+    void testJwtBearerOAuth2AuthorizedClientProviderAuthExceptionWhenUsingPrivateKeyJwtMethod() {
+        OAuth2ClientAuthenticationJwkResolver jwkResolver = spy(new TestOAuth2ClientAuthenticationJwkResolver());
+        resourceServerWithOboContextRunner()
+            .withBean(OAuth2ClientAuthenticationJwkResolver.class, () -> jwkResolver)
+            .withUserConfiguration(AadOAuth2ClientConfiguration.class)
+            .withPropertyValues(
+                "spring.cloud.azure.active-directory.enabled=true",
+                "spring.cloud.azure.active-directory.credential.client-certificate-path=/test/test.pfx",
+                "spring.cloud.azure.active-directory.credential.client-certificate-password=test",
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/aad/implementation/oauth2/AadOAuth2ClientConfigurationTests.java:164]: the certificate is not used if `TestOAuth2ClientAuthenticationJwkResolver` is used.

Comment (moarychan) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/aad/implementation/oauth2/AadOAuth2ClientConfigurationTests.java:164]: Yes, I assume here is testing the bean creation , the `OAuth2ClientAuthenticationJwkResolver` test case is this class [AadOAuth2ClientAuthenticationJwkResolverTests](https://github.com/moarychan/azure-sdk-for-java/blob/316164cf154ce24b317d92b02151783b66fcef00/sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/aad/implementation/oauth2/AadOAuth2ClientAuthenticationJwkResolverTests.java#L28)

**Comment by moarychan** [comment]
Hi @saragluna @chenrujun please help to review!

**Review by chenrujun — APPROVED** [review]
LGTM.

**Review by saragluna — APPROVED** [review]

**Comment by saragluna** [comment]
Should we update the CHANGELOG as well? @moarychan

**Review by saragluna — APPROVED** [review]

**Pull request merged**